pkg/migration-tool/reader/remote_reader.go

```diff
@@ -18,34 +16,38 @@ import (
 	"github.com/timescale/promscale/pkg/migration-tool/utils"
 )
 
-const defaultReadTimeout = time.Minute * 5
-
 type RemoteRead struct {
-	c               context.Context
-	url             string
-	client          *utils.Client
-	plan            *plan.Plan
-	fetchersRunning atomic.Uint32
-	sigBlockRead    chan *plan.Block // To the writer.
-	SigForceStop    chan struct{}
+	c            context.Context
+	url          string
+	client       *utils.Client
+	plan         *plan.Plan
+	sigBlockRead chan *plan.Block // To the writer.
+	sigForceStop chan struct{}
 }
 
-// New creates a new RemoteRead. It creates a ReadClient that is imported from Prometheus remote storage.
+// NewRemoteRead creates a remote-reader. It creates a ReadClient that is imported from Prometheus remote storage.
 // RemoteRead takes help of plan to understand how to create fetchers.
-func New(c context.Context, readStorageUrl string, p *plan.Plan, sigRead chan *plan.Block) (*RemoteRead, error) {
-	rc, err := utils.NewClient(fmt.Sprintf("reader-%d", 1), readStorageUrl, utils.Read, model.Duration(defaultReadTimeout))
+func NewRemoteRead(c context.Context, url string, p *plan.Plan, sigRead chan *plan.Block, needsStopChan bool) (Reader, error) {
+	var reader *RemoteRead
+	rc, err := utils.NewClient(fmt.Sprintf("reader-%d", 1), url, utils.Read, model.Duration(defaultReadTimeout))
 	if err != nil {
 		return nil, fmt.Errorf("creating read-client: %w", err)
 	}
-	read := &RemoteRead{
+	reader = &RemoteRead{
 		c:            c,
-		url:          readStorageUrl,
+		url:          url,
 		plan:         p,
 		client:       rc,
 		sigBlockRead: sigRead,
 	}
-	read.fetchersRunning.Store(0)
-	return read, nil
+	if needsStopChan {
+		reader.sigForceStop = make(chan struct{})
+	}
+	return reader, nil
+}
+
+func (rr *RemoteRead) SigStop() {
+	rr.sigForceStop <- struct{}{}
```

Comment:
I think you need a if not nil check here? Otherwise this will block? or do we want that?

Reply:
This is just used during testing. But, I agree as it will save time during debugging in-case something goes wrong near this place (as non-initialized channels never strike the mind during debugging).